Objects/longobject.c

```diff
@@ -4231,8 +4262,30 @@ long_pow(PyObject *v, PyObject *w, PyObject *x)
         c = (PyLongObject *)x;
         Py_INCREF(x);
     }
-    else if (x == Py_None)
+    else if (x == Py_None) {
         c = NULL;
+        if (a == DIGIT_TWO) {
```

Comment:
Same here: it would be better to check that the value of `a` is 2. Then that catches weird cases where `a == 2` but `a` is not _the_ small integer `2`. Arguably those cases shouldn't exist, but there were some that were recently found and fixed, and it's rather likely that there are more (or that future changes will inadvertently introduce more).

Reply:
Fixed.